app/models/notification.rb

```diff
@@ -41,6 +45,12 @@ def seen_by_all_recipients?
     notification_recipients.unseen.empty?
   end
 
+  def self.notification_text(event_type, full_data)
+    return unless NotificationType.names.include?(event_type) && full_data
```

Comment:
Can you avoid the `NotificationType.names` lookup when `full_data` is nil by swapping the order of this check?